addons/mail/wizard/mail_template_preview.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class MailTemplatePreview(models.TransientModel):
+    _inherit = "mail.template"
+    _name = "mail.template.preview"
+    _description = "Email Template Preview"
+
+    @api.model
+    def _selection_target_model(self):
+        return [(model.model, model.name) for model in self.env['ir.model'].search([])]
+
+    @api.model
+    def _selection_languages(self):
+        return self.env['res.lang'].get_installed()
+
+    @api.model
+    def default_get(self, fields):
+        result = super(MailTemplatePreview, self).default_get(fields)
+        default_mail_template_id = self.env.context.get('default_mail_template_id')
+        if not default_mail_template_id:
+            return result
+        mail_template = self.env['mail.template'].browse(self.env.context['default_mail_template_id'])
+        res = self.env[mail_template.model_id.model].search([], limit=1)
+        if res and 'resource_ref' in fields:
+            result['resource_ref'] = '%s,%s' % (mail_template.model_id.model, res.id)
+        return result
+
+    mail_template_id = fields.Many2one('mail.template', string="Related Mail Template")
+    model_id = fields.Many2one('ir.model', string='Targeted model', related="mail_template_id.model_id")
+    resource_ref = fields.Reference(string='Record reference', selection='_selection_target_model')
+    lang = fields.Selection(_selection_languages, string='Template Preview Language')
+    # Overide fields from mail.template model
+    attachment_ids = fields.Many2many(string='Attachments', store=False)
```

Comment:
`attachment_ids` is indeed an override but `partner_ids` doesn't seem to be.
Maybe separate fields that are overrides and others with an empty space? It's a bit confusing.